src/tree.rs

```diff
@@ -169,7 +169,15 @@ impl<Tab> Node<Tab> {
     #[inline]
     pub fn remove_tab(&mut self, tab_index: TabIndex) -> Option<Tab> {
         match self {
-            Node::Leaf { tabs, .. } => Some(tabs.remove(tab_index.0)),
+            Node::Leaf { tabs, active, .. } => {
+                let tab = tabs.remove(tab_index.0);
+
+                if tab_index <= *active {
+                    active.0 = active.0.saturating_sub(1);
+                }
+
+                Some(tab)
```

Comment:
Any reason we don't call `tabs.remove` here instead?

Reply:
No reason, I fixed it.